lib/f_service/result/base.rb

```diff
@@ -122,7 +125,7 @@ def on_failure(*target_types, unhandled: false)
       def to_ary
         data = successful? ? value : error
 
-        [data, type]
+        [data, @matching_types.first]
```

Comment:
I may be wrong, but it seems to me that the order of the types arguments do matter. If it does so I think that it should be addressed on the tutorial

Reply:
@andre-filho could you check the README changes, please?